### PR TITLE
vi: control-flow-basics/excercise.md translations

### PR DESCRIPTION
**Pull request by HuyNVuong**
I think the msg in the file maybe out of sync with https://google.github.io/comprehensive-rust/control-flow-basics/exercise.html.

Should I continue to refer to the .po file as a source-of-truth, or should I update the English message, as well as the Vietnamese translation in this file?

Or should we re-sync the vi.po file first?

### CODE REVIEW

File: po/vi.po [hunk 1]
```diff
@@ -3584,168 +3584,172 @@ msgid ""
 "The [Collatz Sequence](https://en.wikipedia.org/wiki/Collatz_conjecture) is "
 "defined as follows, for an arbitrary n"
 msgstr ""
+"[Chuỗi Collatz](https://vi.wikipedia.org/wiki/"
+"Gi%E1%BA%A3_thuy%E1%BA%BFt_Collatz) được định nghĩa như sau: với một giá trị "
```
Comment (HuyNVuong) [inline po/vi.po:3588]: I gave the wiki link in Vietnamese instead, is that ok?

**Comment by daivinhtran** [comment]
@mgeisler do you have any thought?

**Review by qu-ngx — COMMENTED** [review]
@daivinhtran he already added the link to the msgstr. Do you want to make any other changes?

File: po/vi.po (hunk 1 above)
Comment (mgeisler) [reply po/vi.po:3588]: I think that's great! This flexibility is exactly why I haven't gone to the trouble of removing the links from the translation data.

**Comment by mgeisler** [comment]
> I think the msg in the file maybe out of sync with https://google.github.io/comprehensive-rust/control-flow-basics/exercise.html.
> 
> Should I continue to refer to the .po file as a source-of-truth, or should I update the English message, as well as the Vietnamese translation in this file?

You should translate based on the data in the PO file. The actual build process is a bit more complex than running `mdbook build`: you can see it in [`build.sh`](https://github.com/google/comprehensive-rust/blob/main/.github/workflows/build.sh).

Most importantly, we run `git restore` on the `src/` directory before generating each translation. We restore the files back to their state on the date recorded in the `POT-Creation-Date` header of your PO file.

The effect of this is that each translation is frozen in time. So even though the English text keeps changing (typos are fixed, slides are added and removed), the Vietnamese translation won't see these changes until you sync the translation.

> Or should we re-sync the vi.po file first?

Syncing the translation is always appreciated! We now have a [translation report](https://google.github.io/comprehensive-rust/translation-report.html) and this shows that the translation was last refreshed on 2024-05-21. That's very recent, but the course keeps evolving, so feel free to refresh again to include the latest changes.

**Comment by mgeisler** [comment]
I think we can merge this one?

**Review by qu-ngx — APPROVED** [review]
@mgeisler okay SGTM. I will merge the PR for now.

**Comment by mgeisler** [comment]
> @mgeisler okay SGTM. I will merge the PR for now.

Thanks a lot! I've been trying to get some of the stale translation PRs over the finishing line. It's hard work to make these PRs and it's tedious work to review them... difficult for all parties. So I hope we can can help things along by merging PRs when they are "good enough".

**Pull request merged**